public/components/overview/mitre_attack_intelligence/intelligence.tsx

```diff
@@ -12,62 +12,77 @@
  */
 
 import React, { useCallback, useState, useRef, useEffect } from 'react';
-import {
-  EuiFlexGroup,
-  EuiFlexItem,
-} from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 
 import { MitreAttackResources } from './resources';
 import { ModuleMitreAttackIntelligenceLeftPanel } from './intelligence_left_panel';
 import { ModuleMitreAttackIntelligenceRightPanel } from './intelligence_right_panel';
 import { useAsyncAction } from '../../common/hooks';
 import { WzRequest } from '../../../react-services';
 import { PanelSplit } from '../../common/panels';
+import { withUserAuthorizationPrompt } from '../../common/hocs';
+import { compose } from 'redux';
 
-export const ModuleMitreAttackIntelligence = () => {
+const ModuleMitreAttackIntelligenceComponent = () => {
```

Comment:
**suggestion**: we could define the component to use directly including the HOC and not create a intermediate component.

```tsx
export const ModuleMitreAttackIntelligence = compose(
  withUserAuthorizationPrompt([{ action: 'mitre:read', resource: '*:*:*' }])
)(() => {
  /* Component definition */
});
```

Reply:
Removed intermediate component and added tests to the component composed with permissions hoc